module/documents/actor/actor.mjs

```diff
@@ -1488,15 +1488,27 @@ export default class Actor5e extends Actor {
       if ( Number.isNumeric(abilityValue) ) parts.push(String(abilityValue / 100));
     }
 
-    // Create the d20 roll
-    const formula = parts.join(" + ");
-    return new CONFIG.Dice.D20Roll(formula, data, {
-      flavor: flavor ?? game.i18n.localize("DND5E.Initiative"),
-      advantageMode: advantageMode,
-      halflingLucky: flags.halflingLucky ?? false,
+    foundry.utils.mergeObject(options, {
```

Comment:
I believe this merge should be reversed, so options provided into the method take priority over the ones set here. That is how other rolling methods are set up:

```javascript
const rollData = foundry.utils.mergeObject({
  flavor: options.flavor ?? game.i18n.localize("DND5E.Initiative"),
  halflingLuck: flags.halflingLucky ?? false,
  critical: null,
  fumble: null
}, options);
```

Reply:
Had a look at the other rolls, and this should be `halflingLucky`, not `halflingLuck`.

Reply:
Good catches, thanks